File: ios-SpaceSub/ios-SpaceSub/Models/Forecast.swift
import Foundation

nonisolated struct ForecastResponse: Codable, Sendable {
    let next7DaysTotal: Double
    let next30DaysTotal: Double
    let next12MonthsTotal: Double
    let upcomingTimeline: [TimelineEntry]
}

nonisolated struct TimelineEntry: Codable, Identifiable, Sendable {
    let merchant: String
    let amount: Double
    let chargeDate: String
    let periodType: BillingCycle

    var id: String { "\(merchant)-\(chargeDate)" }
}

File: android-SpaceSub/app/src/main/java/dev/squad52/spacesub/models/Forecast.kt
package dev.squad52.spacesub.models

data class ForecastResponse(
    val next7DaysTotal: Double,
    val next30DaysTotal: Double,
    val next12MonthsTotal: Double,
    val upcomingTimeline: List<TimelineEntry>
)

data class TimelineEntry(
    val merchant: String,
    val amount: Double,
    val chargeDate: String,
    val periodType: BillingCycle
)
